### ISSUE TITLE
can't define matcher?

### ISSUE THREAD

**Issue by swadey** [andrewcooke/ParserCombinator.jl#2]
I'm trying to define a grammar with a loop, so I'm using `Delayed()`.  I get an error when trying to redefine the matcher for a delayed rule:
```
ERROR: `convert` has no method matching convert(::Type{Nullable{Matcher}}, ::Alt)
```
Here's the grammar:
```julia
expr         = Delayed()                                                                                                                                                        
doubley      = p"[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?[dD]" > (x -> float64(x[1:end-1]))                                                                                       
floaty_dot   = p"[-+]?[0-9]*\.[0-9]+([eE][-+]?[0-9]+)?[Ff]" > (x -> float32(x[1:end-1]))                                                                                        
floaty_nodot = p"[-+]?[0-9]*[0-9]+([eE][-+]?[0-9]+)?[Ff]" > (x -> float32(x[1:end-1]))                                                                                          
floaty       = floaty_dot | floaty_nodot                                                                                                                                        
expr.matcher = doubley | float
```
I'm basically copying what you have in `calc.jl`

**Comment by andrewcooke** [comment]
yeah, it's related to https://github.com/JuliaLang/Compat.jl/issues/94 - it works on 0.4, but not in 0.3 (even with Compat).

however, there's a simple workaround - see https://github.com/andrewcooke/ParserCombinator.jl/blob/213a7f39d875aefcb7b5ff0751277b7b793e1208/test/calc.jl#L26

that's not in the current code / docs (sorry), which are failing.  but it will be soon (hopefully end of today).

also, the code has changed a lot in the last few days.  you might be best either following trunk or doing Pkg.update() regularly.  i had no idea people were actually using it, so i'll push out new releases more often.  thanks!

**Comment by andrewcooke** [comment]
i'll leave this open until that issue is closed.

**Comment by swadey** [comment]
Yeah, I ended up doing this too.  Thanks for the help!

**Comment by andrewcooke** [comment]
i just pushed 1.1.0 which is a fairly big improvement on 1.0.0 if you're using that.  hopefully i haven't broken anything - i don't think there was much change to the externally visible api.  but the new debugger is really useful if you're trying to understand why a grammar doesn't "work" and i've also improved the calc grammar used in the example.

**Comment by swadey** [comment]
I was on master (cloned).  Looks like there's no issue for me with the latest version.

**Comment by andrewcooke** [comment]
i dont think this is goign to be fixed on old releases, and now we have 0.4.
